bolt/lib/Rewrite/DWARFRewriter.cpp

```diff
@@ -613,48 +613,56 @@ void DWARFRewriter::updateDebugInfo() {
   }
 
   if (BC.isDWARF5Used()) {
-    AddrWriter = std::make_unique<DebugAddrWriterDwarf5>(&BC);
+    FinalAddrWriter = std::make_unique<DebugAddrWriterDwarf5>(&BC);
```

Comment:
Since we are now passing instances of AddressWriter to update functions let's make FinalAddrWriter local to this function and pass it to finalize functions where it's used.
This way when this function exits we free up memory automatically.